agent/structs/config_entry_ingress.go

```diff
@@ -0,0 +1,130 @@
+package structs
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/consul/acl"
+)
+
+// IngressGatewayConfigEntry manages the configuration for an ingress service
+// with the given name.
+type IngressGatewayConfigEntry struct {
+	Kind string
+	Name string
+
+	Listeners []IngressListener
+
+	EnterpriseMeta `hcl:",squash" mapstructure:",squash"`
+	RaftIndex
+}
+
+type IngressListener struct {
+	Port     int
+	Protocol string
+
+	Services []IngressService
+}
+
+type IngressService struct {
+	Name          string
+	Namespace     string
```

Comment:
Should this be another `EnterpriseMeta`? That way if a user attempts to set a config entry in oss with a namespace they will get an immediate error as opposed to allowing them to insert data that will not be used.